velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -739,6 +739,18 @@ TEST_F(StringTest, conv) {
   EXPECT_EQ(conv("-10", 16, -10), "-16");
 
   // Overflow case.
+  EXPECT_EQ(
+      conv("-9223372036854775809", 10, -2),
+      "-111111111111111111111111111111111111111111111111111111111111111");
+  EXPECT_EQ(
```

Comment:
Hi @ulysses-you, I found vanilla spark reported an exception for the below test cases you added. Could you confirm again?
```sql
SELECT conv("-9223372036854775808", 10, -2);
SELECT conv("9223372036854775808", 10, -2);
SELECT conv("8000000000000000", 16, -2);
```
Vanilla spark log:
```txt
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 6.0 failed 1 times, most recent failure: Lost task 0.0 in stage 6.0 (TID 6) (sr597 executor driver): java.lang.ArrayIndexOutOfBoundsException: -1
        at org.apache.spark.sql.catalyst.util.NumberConverter$.convert(NumberConverter.scala:163)
        at org.apache.spark.sql.catalyst.util.NumberConverter.convert(NumberConverter.scala)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificUnsafeProjection.apply(Unknown Source)
        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificUnsafeProjection.apply(Unknown Source)
```
```

Reply:
it has been fixed in master branch, see https://github.com/apache/spark/pull/43880